openedx/adg/lms/applications/forms.py

```diff
@@ -108,3 +108,26 @@ def __init__(self, *args, **kwargs):
         self.fields['phone_number'].required = True
         self.fields['city'].disabled = True
         self.fields['name'].disabled = True
+
+
+class UserApplicationAdminForm(forms.ModelForm):
+    """
+    Extend form for UserApplication ADG admin view.
+
+    Extension is required to add a validation check ensuring that the application review form cannot be submitted by an
+    admin unless a decision is made regarding the status of the application.
+    """
+
+    class Meta:
+        model = UserApplication
+        fields = '__all__'
+        help_texts = {'cover_letter_file': None, 'resume': None}
+
+    def __init__(self, *args, **kwargs):
+        self.request = kwargs.pop('request')
+        super(UserApplicationAdminForm, self).__init__(*args, **kwargs)
+
+    def clean(self):
+        super(UserApplicationAdminForm, self).clean()
+        if 'status' not in self.request.POST:
+            raise forms.ValidationError(_('Please make a decision before submitting.'))
```

Comment:
Can we get the `cleaned_data` from the super and then check if status is in cleaned_data and then we should return cleaned_data at the end of this method as it works like this. Any specific reasons for not using cleaned_data from the super call and then not return at the end as well?

Reply:
@asadali145 the status won't be in the cleaned_data. The reason is that we are not using the status field from the model in the form, rather, if you look at the theme PR, you'll see that we have created a custom html field to populate status:

```
<label>
    <input type="radio" id="accept" name="status" value="accepted">
    {{ _('Accept') }}
</label>
```

We are checking for this status in the request object as this status is actually coming from the request object instead of self.cleaned_data. In fact, self.cleaned_data will always be an empty dictionary for this form, therefore there is no point in returning it. All the data is purely readonly for the admin, except two fields i.e. status and admin notes, and for these we have defined custom html.

Reply:
thank you for the reply, Got it. I have not been through theme PR. Sorry for that.